File: gm-utils/src/main/java/src/commom/utils/string/StringCorretorOrtografico.java
package src.commom.utils.string;

import js.support.console;

public class StringCorretorOrtografico {

	private static final String cao = StringConstants.cedilha + StringConstants.a_til + "o";
	private static final String coes = StringConstants.cedilha + StringConstants.o_til + "es";
	private static final String orio = StringConstants.o_agudo + "rio";
	private static final String ario = StringConstants.a_agudo + "rio";
	private static final String itona = StringConstants.i_agudo + "tona";

	public static String exec(String s) {

		if (StringEmpty.is(s)) {
			return s;
		}

		s = StringReplacePalavra.exec(s, "eh", StringConstants.e_agudo);
		s = StringReplacePalavra.exec(s, "Eh", StringConstants.E_agudo);

		StringBox box = new StringBox(s);
		box.replace("nvalid", "nv"+StringConstants.a_agudo+"lid");

		StringConstants.SIMBOLOS.forEach(b -> {
			box.replace("cao"+b, cao + b);
			box.replace("coes"+b, coes + b);
			box.replace("orio"+b, orio + b);
			box.replace("ario"+b, ario + b);
			box.replace("itona"+b, itona + b);
		});

		return box.get();

	}

	public static void main(String[] args) {
		console.log(exec("isso eh demais"));
		console.log(exec("Eh isso eh demais"));
	}

}

File: gm-utils/js/src/commom/utils/string/StringCorretorOrtografico.js
import StringBox from './StringBox';
import StringConstants from './StringConstants';
import StringEmpty from './StringEmpty';
import StringReplacePalavra from './StringReplacePalavra';

export default class StringCorretorOrtografico {

	static exec(s) {

		if (StringEmpty.is(s)) {
			return s;
		}

		s = StringReplacePalavra.exec(s, "eh", StringConstants.e_agudo);
		s = StringReplacePalavra.exec(s, "Eh", StringConstants.E_agudo);

		let box = new StringBox(s);
		box.replace("nvalid", "nv"+StringConstants.a_agudo+"lid");

		StringConstants.SIMBOLOS.forEach(b => {
			box.replace("cao"+b, StringCorretorOrtografico.cao + b);
			box.replace("coes"+b, StringCorretorOrtografico.coes + b);
			box.replace("orio"+b, StringCorretorOrtografico.orio + b);
			box.replace("ario"+b, StringCorretorOrtografico.ario + b);
			box.replace("itona"+b, StringCorretorOrtografico.itona + b);
		});

		return box.get();

	}

	static main(args) {
		console.log(StringCorretorOrtografico.exec("isso eh demais"));
		console.log(StringCorretorOrtografico.exec("Eh isso eh demais"));
	}

}
StringCorretorOrtografico.cao = StringConstants.cedilha + StringConstants.a_til + "o";
StringCorretorOrtografico.coes = StringConstants.cedilha + StringConstants.o_til + "es";
StringCorretorOrtografico.orio = StringConstants.o_agudo + "rio";
StringCorretorOrtografico.ario = StringConstants.a_agudo + "rio";
StringCorretorOrtografico.itona = StringConstants.i_agudo + "tona";
